libtransact/src/contract/archive/mod.rs

```diff
@@ -325,6 +345,25 @@ mod tests {
         assert!(find_scar("nonexistent", "0.1.0", &[&dir]).is_err());
     }
 
+    /// Verify that an error is returned when the contract name contains '_'.
```

Comment:
For these two tests, you should write a matching scar file with `write_mock_scar` to ensure that the error is being returned because the name/version is invalid, not because the file doesn't exist.

Reply:
Addressed